### DISCUSSION TITLE
Bug: Text field integer is being rounded and displaying incorrectly

### DISCUSSION THREAD

**Discussion by bkuhl** [laravel/nova-issues#6071] [Q&A]
- Laravel Version: 10.32.1
- Nova Version: 4.31.3
- PHP Version: 8.2.3
- Database Driver & Version: MySQL 8.0.28
- Operating System and Version: #.#
- Browser type and version: Chrome Version 119.0.6045.159 (Official Build) (arm64)
- Reproduction Repository: https://github.com/###/###

### Description:

I've noticed when displaying large numbers as Text that the numbers are being altered somewhere in the frontend Javascript leading to incorrect values being displayed.  Searching a resource by this number and updating the value for this number via a form are saved correctly, it seems to only be a display issue.  

### Detailed steps to reproduce the issue on a fresh Nova installation:

1) Create a table with the following structure:

```
CREATE TABLE `channel_subscriptions` (
  `id` bigint unsigned NOT NULL AUTO_INCREMENT,
  `platform_thread_id` bigint unsigned DEFAULT NULL,
  PRIMARY KEY (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
```
2) Create a nova resource for this table with the field:

```
Text::make('Platform Thread Id','platform_thread_id'),
```

3) Add a record to the table with `platform_thread_id` set to value `1120706941212172299`.

Showing the resource on Nova displays the number as `1120706941212172300` which is incorrect.  I handle this as a Text field in Nova expecting that it would avoid any Javascript rounding issues.  This incorrect number also shows when editing the resource from a relation (as shown below).  Re-entering the correct value and saving this form saves the correct value.

<img width="1049" alt="Screenshot 2023-11-28 at 9 29 02 AM" src="https://github.com/laravel/nova-issues/assets/524933/48a158bc-9bf3-4aa1-9f06-c8a2168995b1">

As well as when showing the resource via an index page (again using it as a Text field):

<img width="262" alt="Screenshot 2023-11-28 at 9 30 02 AM" src="https://github.com/laravel/nova-issues/assets/524933/637fe1d6-2edd-461d-b531-523d3e6c0745">

I can also see in the ajax requesting loading the data that it's actually loading the correct value, so the value appears to be altered somewhere in the frontend JS before it's displayed:

<img width="272" alt="Screenshot 2023-11-28 at 9 31 17 AM" src="https://github.com/laravel/nova-issues/assets/524933/39c0c0ae-0840-4472-b335-04a48c5652ce">

> screenshot taken from ajax request `/nova-api/channel-subscriptions?search=Channel&filters=W3siQXBwXFxOb3ZhXFxGaWx0ZXJzXFxDaGFubmVsU3Vic2NyaXB0aW9uc1xcQnlDYXRlZ29yeSI6IiJ9LHsiQXBwXFxOb3ZhXFxGaWx0ZXJzXFxDaGFubmVsU3Vic2NyaXB0aW9uc1xcQnlDaGFubmVsVHlwZSI6IiJ9XQ%3D%3D&orderBy=&perPage=15&trashed=&page=1&viaResource=server&viaResourceId=875707&viaRelationship=channelSubscriptions&relationshipType=hasMany`

**Comment by crynobone** [comment]
`Text` field was design to handle string, in order 5o fix this you can use `displayUsing()` and casting the value to `string`.

**Comment by bkuhl** [comment]
Yep, that fixed it.  Thanks.
